tmva/tmva/src/MethodBDT.cxx

```diff
@@ -2765,50 +2765,93 @@ void TMVA::MethodBDT::MakeClassSpecific( std::ostream& fout, const TString& clas
    fout << "   std::vector<"<<nodeName<<"*> fForest;       // i.e. root nodes of decision trees" << std::endl;
    fout << "   std::vector<double>                fBoostWeights; // the weights applied in the individual boosts" << std::endl;
    fout << "};" << std::endl << std::endl;
-   fout << "double " << className << "::GetMvaValue__( const std::vector<double>& inputValues ) const" << std::endl;
-   fout << "{" << std::endl;
-   fout << "   double myMVA = 0;" << std::endl;
-   if (fDoPreselection){
-      for (UInt_t ivar = 0; ivar< fIsLowBkgCut.size(); ivar++){
-         if (fIsLowBkgCut[ivar]){
-            fout << "   if (inputValues["<<ivar<<"] < " << fLowBkgCut[ivar] << ") return -1;  // is background preselection cut" << std::endl;
-         }
-         if (fIsLowSigCut[ivar]){
-            fout << "   if (inputValues["<<ivar<<"] < "<< fLowSigCut[ivar] << ") return  1;  // is signal preselection cut" << std::endl;
-         }
-         if (fIsHighBkgCut[ivar]){
-            fout << "   if (inputValues["<<ivar<<"] > "<<fHighBkgCut[ivar] <<")  return -1;  // is background preselection cut" << std::endl;
-         }
-         if (fIsHighSigCut[ivar]){
-            fout << "   if (inputValues["<<ivar<<"] > "<<fHighSigCut[ivar]<<")  return  1;  // is signal preselection cut" << std::endl;
+
+   if(GetAnalysisType() == Types::kMulticlass) {
+      fout << "std::vector<Float_t>& ReadBDTG::GetMulticlassValues__( const std::vector<double>& inputValues ) const" << std::endl;
+      fout << "{" << std::endl;
+      fout << "   std::vector<Float_t> *fMulticlassReturnVal = new std::vector<Float_t>();" << std::endl;
```

Comment:
We know the desired size of the vector here, and we know that we will fill every element so I believe the preferred approach is
```
std::vector<Float_t> multiclassReturnVal;
multiclassReturnVal.reserve(nClasses);
```